godot-cell/src/lib.rs

```diff
@@ -128,6 +188,25 @@ impl<T> GdCell<T> {
 
         state.borrow_state.shared_count() > 0 || state.borrow_state.mut_count() > 0
     }
+
+    #[cfg(feature = "threads")]
+    fn claim_mut_thread<'g>(
+        &'g self,
+        mut_thread_id: RwLockReadGuard<'g, std::thread::ThreadId>,
+    ) -> RwLockReadGuard<'g, std::thread::ThreadId> {
+        if *mut_thread_id == std::thread::current().id() {
+            mut_thread_id
+        } else {
+            drop(mut_thread_id);
+
+            let mut mut_thread_id = self.mut_thread.write();
+
+            *mut_thread_id = std::thread::current().id();
+
+            // get new read lock so no one can write while we hand out guards
```

Comment:
Use sentences when possible:
```suggestion
            // Get new read lock, so no one can write while we hand out guards.
```